internal/distributed/proxy/httpserver/utils.go

```diff
@@ -0,0 +1,844 @@
+		case schemapb.DataType_BinaryVector:
+			rowsNum = int64(len(fieldDataList[0].GetVectors().GetBinaryVector())*8) / fieldDataList[0].GetVectors().GetDim()
+		case schemapb.DataType_FloatVector:
+			rowsNum = int64(len(fieldDataList[0].GetVectors().GetFloatVector().Data)) / fieldDataList[0].GetVectors().GetDim()
+		default:
+			return nil, fmt.Errorf("the type(%v) of field(%v) is not supported, use other sdk please", fieldDataList[0].Type, fieldDataList[0].FieldName)
+		}
+	}
+	for i := int64(0); i < rowsNum; i++ {
+		row := map[string]interface{}{}
+		for j := 0; j < columnNum; j++ {
+			switch fieldDataList[j].Type {
+			case schemapb.DataType_Bool:
+				row[fieldDataList[j].FieldName] = fieldDataList[j].GetScalars().GetBoolData().Data[i]
+			case schemapb.DataType_Int8:
+				row[fieldDataList[j].FieldName] = int8(fieldDataList[j].GetScalars().GetIntData().Data[i])
+			case schemapb.DataType_Int16:
+				row[fieldDataList[j].FieldName] = int16(fieldDataList[j].GetScalars().GetIntData().Data[i])
+			case schemapb.DataType_Int32:
+				row[fieldDataList[j].FieldName] = fieldDataList[j].GetScalars().GetIntData().Data[i]
+			case schemapb.DataType_Int64:
+				row[fieldDataList[j].FieldName] = fieldDataList[j].GetScalars().GetLongData().Data[i]
+			case schemapb.DataType_Float:
+				row[fieldDataList[j].FieldName] = fieldDataList[j].GetScalars().GetFloatData().Data[i]
+			case schemapb.DataType_Double:
+				row[fieldDataList[j].FieldName] = fieldDataList[j].GetScalars().GetDoubleData().Data[i]
+			case schemapb.DataType_String:
+				row[fieldDataList[j].FieldName] = fieldDataList[j].GetScalars().GetStringData().Data[i]
+			case schemapb.DataType_VarChar:
+				row[fieldDataList[j].FieldName] = fieldDataList[j].GetScalars().GetStringData().Data[i]
+			case schemapb.DataType_BinaryVector:
+				row[fieldDataList[j].FieldName] = fieldDataList[j].GetVectors().GetBinaryVector()[i*(fieldDataList[j].GetVectors().GetDim()/8) : (i+1)*(fieldDataList[j].GetVectors().GetDim()/8)]
+			case schemapb.DataType_FloatVector:
+				row[fieldDataList[j].FieldName] = fieldDataList[j].GetVectors().GetFloatVector().Data[i*fieldDataList[j].GetVectors().GetDim() : (i+1)*fieldDataList[j].GetVectors().GetDim()]
+			case schemapb.DataType_Array:
+				row[fieldDataList[j].FieldName] = fieldDataList[j].GetScalars().GetArrayData().Data[i]
+			case schemapb.DataType_JSON:
+				data, ok := fieldDataList[j].GetScalars().Data.(*schemapb.ScalarField_JsonData)
+				if ok && !fieldDataList[j].IsDynamic {
+					row[fieldDataList[j].FieldName] = data.JsonData.Data[i]
+				} else {
+					var dataMap map[string]interface{}
+
+					err := json.Unmarshal(fieldDataList[j].GetScalars().GetJsonData().Data[i], &dataMap)
+					if err != nil {
+						log.Error(fmt.Sprintf("[BuildQueryResp] Unmarshal error %s", err.Error()))
+						return nil, err
+					}
+
+					if containsString(dynamicOutputFields, "*") || containsString(dynamicOutputFields, "$meta") {
```

Comment:
There is no need to handle '*'; the Proxy automatically expands '*' and places the final "output_fields" in the respective fields of milvuspb.QueryResults and schemapb.SearchResultData.